packages/block-library/src/navigation/style.scss

```diff
@@ -626,6 +626,7 @@ button.wp-block-navigation-item__content {
 	border: none;
 	margin: 0;
 	padding: 0;
+	text-transform: inherit;
```

Comment:
This is to fix a separate issue I noticed where the text version of these buttons doesn't inherit the text-transform from the navigation block, e.g. if `text-transform: uppercase` is set on the navigation block, this wasn't being applied to the text label.